src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -107,41 +107,36 @@ export class MongoCollectionTreeItem extends AzureParentTreeItem<IMongoTreeRoot>
 		return new MongoDocumentTreeItem(this, newDocument);
 	}
 
-	executeCommand(name: string, args?: string[]): Thenable<string> | null {
+	public async tryExecuteCommandDirectly(name: string, args?: string[]): Promise<{ deferToShell?: boolean; result?: string }> {
```

Comment:
Does it need At least one of the fields, exactly one of the fields, or is returning an empty object OK too?

Reply:
Now it's very explicit:  Promise<{ deferToShell: true; result: undefined } | { deferToShell: false; result: string }>